### PR TITLE
fixes vender exploit

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
Makes it so that psionic items cant be placed inside custom venders.

### CODE REVIEW

File: code/game/machinery/vending.dm [hunk 1]
```diff
@@ -175,6 +175,8 @@
 	var/datum/wires/vending/wires = null
 	var/always_open	=	FALSE // If true, this machine allows products to be inserted without requirinf the maintenance hatch to be screwed open first
 	var/list/can_stock = list()	//A whitelist of objects which can be stocked into this vendor
+	var/list/blacklisted = list() //Items that are not allowed for the vender to use if custom
```
Comment (Smokingsquirrel) [inline code/game/machinery/vending.dm:178]: Vendor not vender

File: code/game/machinery/vending.dm [hunk 2]
```diff
@@ -218,6 +220,11 @@
 	if(!user.unEquip(W))
 		return
 
+	for(var/a in blacklisted)
+		if(istype(W, a))
+			to_chat(user, SPAN_NOTICE("You are unable to add this product to the receptor."))
```
Comment (Smokingsquirrel) [inline code/game/machinery/vending.dm:225]: A receptor/acceptor in vending machines is for the bills. This phrase would imply you add an item through the credit bill slot. The part where you put the item you want to sell/buy is called the product shelf.

The phrase should be "You are unable to add this to the product shelves."

File: code/game/machinery/vending.dm [hunk 3]
```diff
@@ -981,6 +988,27 @@
...
+						/obj/item/tool/psionic_omnitool,
+						/obj/item/projectile/kinetic_blast,
+						/obj/item/device/lighting/toggleable/lantern/psionics,
+						/obj/item/flame/pyrokinetic_spark,
+						/obj/item/tool/hammer/telekinetic_fist,
+						/obj/item/tool/knife/psionic_blade,
+						/obj/item/shield/riot/crusader/psionic,
+						/obj/item/stack/medical/bruise_pack/psionic,
+
+	) //Items that are not allowed for the vender to use if custom
```
Comment (Smokingsquirrel) [inline code/game/machinery/vending.dm:1010]: Vendor, not vender

File: code/game/machinery/vending.dm (hunk 2 above)
Comment (Trilbyspaceclone) [reply code/game/machinery/vending.dm:225]: This is accually same message as the one that rejects items?

Comment (Trilbyspaceclone) [reply code/game/machinery/vending.dm:225]: `to_chat(user, SPAN_NOTICE("You insert \the [W] in the product receptor."))`

Comment (Smokingsquirrel) [reply code/game/machinery/vending.dm:225]: Well then its wrong as well.

**Pull request merged**